tests/test_dataframe_type/test_contains.py

```diff
@@ -159,4 +159,50 @@ def test_does_not_contain_case_insensitive(self):
         self.assertTrue(DataframeType({"value": df}).does_not_contain_case_insensitive({
             "target": "var4",
             "comparator": "var2"
-        }).equals(pandas.Series([False, True, True])))
\ No newline at end of file
+        }).equals(pandas.Series([False, True, True])))
+
+    def test_contains_all(self):
+        df = pandas.DataFrame.from_dict(
+            {
+                "var1": ['test', 'value', 'word'],
+                "var2": ["test", "value", "test"],
+                "variable_names": ["STUDYID", "USUBJID", "COOLVAR"],
+                "required_variables": [["STUDYID", "USUBJID"], ["STUDYID", "USUBJID"], ["STUDYID", "USUBJID"]]
+            }
+        )
+        self.assertTrue(DataframeType({"value": df}).contains_all({
```

Comment:
While I think list flattening is good to have for `contains_all`, I'm not understanding why we need it for CG0014.

In your description, you have:
```yaml
Rule Type: Variable Metadata Check
Check:
   - all
       name: "variable_names"
       operator: "contains_all"
       value:
            - required_variables  # Result of the required variables operations
```

but why didn't it already work with this:

```yaml
Rule Type: Variable Metadata Check
Check:
   - all
       name: "variable_names"
       operator: "contains_all"
       value: required_variables  # Result of the required variables operations
```

I thought list flattening is when you need something like:

```yaml
Rule Type: Variable Metadata Check
Check:
   - all
       name: "variable_names"
       operator: "contains_all"
       value: 
         - "REQVAR1"
         - "REQVAR2"
         - required_variables # Result of the required variables operations
```

Reply:
It didn't work before because the contains_all method neither flattens a list nor checks if the column provided is a column of iterables. The other contains operators check if the column provided is a column of iterables and handles that case appropriately. Because the contains_all did neither, I figured adding the list flattening would cover all use cases.